### PR TITLE
Change PodSecurityPolicy field names.

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by erictune** [comment]
@thockin you asked for this
@pweil- please review too.
We can get this in before 1.2 is cut and there is no api breakage.

**Comment by pweil-** [comment]
LGTM

**Comment by pweil-** [comment]
@pmorie - FYI

**Comment by ncdc** [comment]
unit/int failure:

```
+ godep go install ./...
# k8s.io/kubernetes/pkg/api/testing
/go/src/k8s.io/kubernetes/pkg/api/testing/fuzzer.go:412: psp.RunAsUser.Type undefined (type extensions.RunAsUserStrategyOptions has no field or method Type)
godep: go exit status 2
```

@erictune

**Comment by erictune** [comment]
Thanks Andy, fixed.

File: pkg/apis/extensions/types.go [hunk 1]
```diff
@@ -845,8 +845,8 @@ type PodSecurityPolicySpec struct {
 	HostPID bool `json:"hostPID,omitempty"`
 	// HostIPC determines if the policy allows the use of HostIPC in the pod spec.
 	HostIPC bool `json:"hostIPC,omitempty"`
-	// SELinuxContext is the strategy that will dictate the allowable labels that may be set.
-	SELinuxContext SELinuxContextStrategyOptions `json:"seLinuxContext,omitempty"`
+	// SELinux is the strategy that will dictate the allowable labels that may be set.
+	SELinux SELinuxStrategyOptions `json:"seLinux,omitempty"`
```
Comment (thockin) [inline pkg/apis/extensions/types.go]: for the json tag, do we want "seLinux (implies 2 words) or "selinux" implies a single proper name?  I've always thought of selinux as the latter, but I yield to RH folks who are obviously more invested in selinux.

File: pkg/apis/extensions/types.go [hunk 2]
```diff
@@ -881,30 +881,30 @@ var (
...
+	Rule SELinuxStrategy `json:"rule"`
 	// seLinuxOptions required to run as; required for MustRunAs
 	// More info: http://releases.k8s.io/HEAD/docs/design/security_context.md#security-context
 	SELinuxOptions *api.SELinuxOptions `json:"seLinuxOptions,omitempty"`
 }
 
-// SELinuxContextStrategyType denotes strategy types for generating SELinux options for a
-// SecurityContext.
-type SELinuxContextStrategy string
+// SELinuxStrategyRule denotes strategy types for generating SELinux options for a
```
Comment (thockin) [inline pkg/apis/extensions/types.go]: The comment says "SELinuxStrategyRule" but the type is actually "SELinuxStrategy"

Comment (thockin) [inline pkg/apis/extensions/types.go]: Same mistake is in the RunAsUserStrategy type defn

**Comment by thockin** [comment]
just a comment nit and a naming optional from me.

**Comment by thockin** [comment]
Thanks!

**Comment by eparis** [comment]
@k8s-bot test this, issue #22435

File: pkg/apis/extensions/types.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/apis/extensions/types.go]: I prefer the latter, I don't know whether there are other examples sufficiently thought through.

Comment (erictune) [inline pkg/apis/extensions/types.go]: Well, the RH guys wrote what was there before, and it was spelled `seLinuxContext`, so I just shortened it.

Comment (erictune) [inline pkg/apis/extensions/types.go]: Okay, I'll make it "selinux".

**Comment by liggitt** [comment]
merged after 1.2 branch was cut, marking for cherry-pick

File: pkg/apis/extensions/types.go (hunk 1 above)
Comment (pmorie) [inline pkg/apis/extensions/types.go]: +1 to `selinux`

**Pull request merged**